setup.py

```diff
@@ -11,9 +11,30 @@
 SRC_DIR = os.path.join(THIS_DIR, "src")
 SNOWPARK_SRC_DIR = os.path.join(SRC_DIR, "snowflake", "snowpark")
 CONNECTOR_DEPENDENCY_VERSION = ">=3.2.0, <4.0.0"
+INSTALL_REQ_LIST = [
+    "setuptools>=40.6.0",
+    "wheel",
+    f"snowflake-connector-python{CONNECTOR_DEPENDENCY_VERSION}",
+    "pyyaml",
+]
+CLOUDPICKLE_REQ_LIST = ["cloudpickle>=1.6.0,<=2.0.0"]
+# this allows us to update the cloudpickle dependency by env var when building package privately
+# without re-releasing package just to update the dependency, example if we want to lower cloudpickle dep to 2.1.0:
+# SNOWFLAKE_IS_PYTHON_RUNTIME_TEST=1 CLOUDPICKLE_PYTHON_311_DEPENDENCY="cloudpickle==2.1.0;python_version~='3.11'" pip install .
+CLOUDPICKLE_PYTHON_311_DEPENDENCY = os.getenv(
```

Comment:
nit: This can go inside `if os.getenv("SNOWFLAKE_IS_PYTHON_RUNTIME_TEST", False):` I think? We could move it out of the if branch the same time as we make 3.11 available public